node/child_process.ts

```diff
@@ -2,299 +2,98 @@
-   * This property represents the child process's stdin.
-   */
-  stdin: Writable | null = null;
-
-  /**
-   * This property represents the child process's stdout.
-   */
-  stdout: Readable | null = null;
-
-  /**
-   * This property represents the child process's stderr.
-   */
-  stderr: Readable | null = null;
-
-  /**
-   * Pipes to this child process.
-   */
-  stdio: [Writable | null, Readable | null, Readable | null] = [
-    null,
-    null,
-    null,
-  ];
-
-  #process!: Deno.Process;
-  #spawned = deferred<void>();
-
-  constructor(
-    command: string,
-    args?: string[],
-    options?: ChildProcessOptions,
-  ) {
-    super();
-
-    const {
-      env = {},
-      stdio = ["pipe", "pipe", "pipe"],
-      shell = false,
-    } = options || {};
-
-    const [
-      stdin = "pipe",
-      stdout = "pipe",
-      stderr = "pipe",
-    ] = normalizeStdioOption(stdio);
-    const cmd = buildCommand(
-      command,
-      args || [],
-      shell,
-    );
-    this.spawnfile = cmd[0];
-    this.spawnargs = cmd;
-
-    try {
-      this.#process = Deno.run({
-        cmd,
-        env,
-        stdin: toDenoStdio(stdin as NodeStdio | number),
-        stdout: toDenoStdio(stdout as NodeStdio | number),
-        stderr: toDenoStdio(stderr as NodeStdio | number),
-      });
-      this.pid = this.#process.pid;
-
-      if (stdin === "pipe") {
-        assert(this.#process.stdin);
-        this.stdin = createWritableFromStdin(this.#process.stdin);
-      }
-
-      if (stdout === "pipe") {
-        assert(this.#process.stdout);
-        this.stdout = createReadableFromReader(this.#process.stdout);
-      }
-
-      if (stderr === "pipe") {
-        assert(this.#process.stderr);
-        this.stderr = createReadableFromReader(this.#process.stderr);
-      }
-
-      this.stdio[0] = this.stdin;
-      this.stdio[1] = this.stdout;
-      this.stdio[2] = this.stderr;
-
-      nextTick(() => {
-        this.emit("spawn");
-        this.#spawned.resolve();
-      });
-
-      (async () => {
-        const status = await this.#process.status();
-        this.exitCode = status.code;
-        this.#spawned.then(async () => {
-          // The 'exit' and 'close' events must be emitted after the 'spawn' event.
-          this.emit("exit", this.exitCode, status.signal ?? null);
-          await this._waitForChildStreamsToClose();
-          this.kill();
-          this.emit("close", this.exitCode, status.signal ?? null);
-        });
-      })();
-    } catch (err) {
-      this._handleError(err);
-    }
+import {
+  ChildProcess,
+  ChildProcessOptions,
+  stdioStringToArray,
+} from "./internal/child_process.ts";
+import { validateString } from "./internal/validators.js";
+import {
+  ERR_CHILD_PROCESS_IPC_REQUIRED,
+  ERR_INVALID_ARG_VALUE,
+} from "./_errors.ts";
+import { process } from "./process.ts";
+
+const denoCompatArgv = [
+  "run",
+  "--compat",
+  "--unstable",
+  "--no-check",
+  "--location=https://example.com",
+  "--allow-all",
```

Comment:
Yeah... this is problematic, ideally we'd use the same permissions as the parent process, but unfortunately we don't have access to these flags.